apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/HiPerfPublisher.cs

```diff
@@ -41,109 +47,117 @@ public abstract class HiPerfPublisher
-
-            /// <summary>
-            /// Instantiate settings, optionally specifying client creation parameters.
-            /// </summary>
-            /// <param name="clientCount">Optional. The number of clients to create.
-            /// Defaults to the CPU count on the machine this is being executed on.</param>
-            /// <param name="publisherSettings">Optional. The settings to use when creating <see cref="Publisher"/> instances.</param>
-            /// <param name="credentials">Optional. Credentials to use then creating <see cref="Publisher"/> instances.
-            /// Defaults to using the default credentials.</param>
-            public Settings(int? clientCount = null, PublisherSettings publisherSettings = null, ChannelCredentials credentials = null)
-            {
-                ClientCount = clientCount;
-                PublisherSettings = publisherSettings;
-                Credentials = credentials;
-            }
+            public Settings() { }
 
             internal Settings(Settings other)
             {
-                Clients = other.Clients != null ? new List<PublisherClient>(other.Clients) : null;
-                ClientCount = other.ClientCount;
-                PublisherSettings = other.PublisherSettings?.Clone();
-                Credentials = other.Credentials;
-                BatchingSettings = other.BatchingSettings?.Clone();
+                BatchingSettings = other.BatchingSettings;
+                MaxBatchingSettings = other.MaxBatchingSettings;
                 Scheduler = other.Scheduler;
-                BatchMaxElementCount = other.BatchMaxElementCount;
-                BatchMaxByteCount = other.BatchMaxByteCount;
             }
 
             /// <summary>
-            /// The clients to use within a <see cref="HiPerfPublisher"/>.
+            /// <see cref="BatchingSettings"/> that control how messages are batched when sending.
+            /// If <c>null</c>, defaults to <see cref="DefaultBatchingSettings"/>.
             /// </summary>
-            public IList<PublisherClient> Clients { get; }
+            public BatchingSettings BatchingSettings { get; set; }
 
             /// <summary>
-            /// The number of <see cref="PublisherClient"/> to create and use within a <see cref="HiPerfPublisher"/> instance.
-            /// If <c>null</c>, defaults to the CPU count on the machine this is being executed on.
+            /// <see cref="BatchingSettings"/> that control how messages are batched when a batch
+            /// cannot be immediately sent (because all clients are already busy sending batches).
+            /// <see cref="BatchingSettings.DelayThreshold"/> is not relevant in this context.
+            /// If <c>null</c>, defaults to <see cref="ApiMaxBatchingSettings"/>.
             /// </summary>
-            public int? ClientCount { get; }
+            public BatchingSettings MaxBatchingSettings { get; set; }
```

Comment:
QueuedBatchingSettings perhaps? (Definitely bike-shedding here - let's not worry about it for now. Just a thought for later.)